sdk/resourcemanager/azure-resourcemanager-appplatform/src/main/java/com/azure/resourcemanager/appplatform/implementation/SpringAppImpl.java

```diff
@@ -113,11 +109,7 @@ public SpringAppDeployment getActiveDeployment() {
 
     @Override
     public Mono<SpringAppDeployment> getActiveDeploymentAsync() {
-        String activeDeploymentName = activeDeploymentName();
-        if (activeDeploymentName == null || activeDeploymentName.isEmpty()) {
-            return Mono.empty();
-        }
-        return deployments().getByNameAsync(activeDeploymentName);
+        return Mono.justOrEmpty(deployments.list().stream().filter(SpringAppDeployment::isActive).findFirst());
```

Comment:
`deployments.list()` is sync, while the method is async?

You cannot call sync in async.

Reply:
Right, changed.